airflow/kubernetes/pod_generator.py

```diff
@@ -301,51 +352,151 @@ def reconcile_pods(base_pod: k8s.V1Pod, client_pod: k8s.V1Pod) -> k8s.V1Pod:
+
+        client_pod_cp.metadata = merge_objects(base_pod.metadata, client_pod_cp.metadata)
+        client_pod_cp = merge_objects(base_pod, client_pod_cp)
 
         return client_pod_cp
+
+    @staticmethod
+    def construct_pod(
+        dag_id: str,
+        task_id: str,
+        pod_id: str,
+        try_number: int,
+        date: str,
+        command: List[str],
+        kube_executor_config: Optional[k8s.V1Pod],
+        worker_config: k8s.V1Pod,
+        namespace: str,
+        worker_uuid: str
+    ) -> k8s.V1Pod:
+        """
+        Construct a pod by gathering and consolidating the configuration from 3 places:
+            - airflow.cfg
+            - executor_config
+            - dynamic arguments
+        """
+
+        dynamic_pod = PodGenerator(
+            namespace=namespace,
+            labels={
+                'airflow-worker': worker_uuid,
+                'dag_id': dag_id,
+                'task_id': task_id,
+                'execution_date': date,
+                'try_number': str(try_number),
+            },
+            cmds=command,
+            name=pod_id
+        ).gen_pod()
+
+        # Reconcile the pod generated by the Operator and the Pod
+        # generated by the .cfg file
+        pod_with_executor_config = PodGenerator.reconcile_pods(worker_config,
+                                                               kube_executor_config)
+        # Reconcile that pod with the dynamic fields.
+        return PodGenerator.reconcile_pods(pod_with_executor_config, dynamic_pod)
+
+    @staticmethod
+    def deserialize_model_file(api_client: ApiClient, path: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param path: Path to the file
+        :return: a kubernetes.client.models.V1Pod
+        """
+        pod = None
+        with open(path) as stream:
+            if '.json' in path:
+                pod = json.load(stream)
+            elif '.yaml' in path:
+                pod = yaml.safe_load(stream)
+            elif not pod:
+                raise AirflowConfigException("Path was neither .json nor .yaml")
+
+            # pylint: disable=protected-access
+            return api_client._ApiClient__deserialize_model(pod, k8s.V1Pod)
+
+    @staticmethod
+    def deserialize_model_string(api_client: ApiClient, string: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param string: a string of the deployment
+        :return: a kubernetes.client.models.V1Pod
+        """
+        try:
+            pod = json.loads(string)
+        except json.decoder.JSONDecodeError:
+            try:
+                pod = yaml.safe_load(string)
+            except ScannerError:
+                raise AirflowConfigException(
+                    "Could not parse {} as yaml or json".format(string)
+                )
+
+        # pylint: disable=protected-access
+        return api_client._ApiClient__deserialize_model(pod, k8s.V1Pod)
```

Comment:
This is a private method. Rather, we should not rely on the existence of this method.

Reply:
Really not great having to do this. Is there no other option?

Reply:
@davlum any word on if we can get around using a private method?

Reply:
Doesn't look like there's been any mouvement on the k8s client, just pinged them again.

Reply:
And you got a response https://github.com/kubernetes-client/python/issues/977#issuecomment-580009695 - is that workable?

At the very least, can you put a doc comment here linking to the issue in the kube repo please?

Reply:
Unfortunately that solution is for modifying objects returned from calling the client, as opposed to going directly from YAML -> client object. The [public deserialize method](https://github.com/kubernetes-client/python/blob/5181d2389eacd92fa0f72237da87286ed0d45a3e/kubernetes/client/api_client.py#L226) deserializes from a RESTResponse object and calls the [private deserialize method](https://github.com/kubernetes-client/python/blob/5181d2389eacd92fa0f72237da87286ed0d45a3e/kubernetes/client/api_client.py#L246) that we need. I've added to the doc string and referenced back to the issue on the kubernetes client.